### ISSUE TITLE
Show Answer button jumps view back to top

### ISSUE THREAD

**Issue by sanjayananda** [ankidroid/Anki-Android#5182]
For context, I primarily have cards that have long texts with clozes. This bug relates to those types of cards.

###### Reproduction Steps

1. Open a deck's cards for study
2. Scroll down until I find the cloze "[...]"
3. Tap "Show Answer"

###### Expected Result

The page view stays where it is - no movement.

###### Actual Result

The page view jumps all the way back to the top of the screen. I have to scroll down again, to find the now-revealed answer.

###### Debug info

AnkiDroid Version = 2.8.4
Android Version = 8.1.0
ACRA UUID = 58cab56b-9a8b-41a8-98ed-91e378e5bc58

###### Research

[ x ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid

[ x ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue

[ x ] I have searched for similar existing issues here and on the user forum

**Comment by mikehardy** [comment]
Hi there and thanks for taking the time to report this. Sorry you've run into trouble. I'm traveling right now so don't have time to test but the first thing I would check is if this happens with the current alpha version. If you follow the instructions here you can check for yourself if you want to help move the issue forward - https://github.com/ankidroid/Anki-Android/wiki/Installation#beta--alpha-releases

**Comment by sanjayananda** [comment]
@mikehardy thanks for this.

I tried to figure out downloading the alpha version, joined the Google group but didn't find an alpha download link there or in the link you shared - it's different than the basic public Play Store version?

I tried the beta version, but the original issue was acting the same, jumping to the top.

**Comment by mikehardy** [comment]
Hi @sanjayananda - The alpha version is much more current than the beta version at this point, and while I don't think it will solve this bug you never know - and it's good to test on it. On the link above, in order to get the Alpha version immediately (after joining the Google alpha testers group) - do this:

```
To get a beta or alpha update with no delay instead of waiting, may manually tap the 
button to scan for updates in the Google Play Store app on your device, then update via 
the update button in the list of app updates found (not in the app details section).
```

**Comment by sanjayananda** [comment]
@mikehardy ok, went through the steps, hope it worked but not clear if it did or not - any way to confirm if I have the alpha version or not?

Issue remains the same.

**Comment by mikehardy** [comment]
In preferences, settings, advanced, at the bottom it has the version

**Comment by mikehardy** [comment]
Strange, but if you see 2.8.4 and you have definitely checked for new versions that means you are probably not in the beta and alpha groups.

**Comment by sanjayananda** [comment]
Hmm, yeah, went through the steps (joined alpha Google group, checked for updates), there was no update as described. Any suggestions?

You mentioned this might not solve the issue anyway - just want to check that this isn't a sidetrack from the scroll jumping issue.

**Comment by mikehardy** [comment]
For now this is a sidetrack but later when this has a fix you won't be able to confirm it until there is a major release - so it is t vital, but maybe useful. Sounds like you tried though for sure! No idea why it didn't work... Maybe your phone is connected to Google play with a different account than the Google alpha group membership

**Comment by sanjayananda** [comment]
Ok, thanks for explaining. Happy to continue down the sidetrack if you see it all connecting 😄

I checked on the different account possibility - both are on the same account.

**Comment by sanjayananda** [comment]
Hey, checking in about this. Any ideas for next steps?

**Comment by mikehardy** [comment]
sorry @sanjayananda - I got roped into a startup in Real Life with some tight deadlines and haven't had time to move any AnkiDroid things forward in the last couple weeks. I've tagged it with needs-triage so it'll stay near the top of the investigation stack but I can't promise when I'll have a chance to look at it, personally :-(

**Comment by sanjayananda** [comment]
Ok thanks Mike. All the best with the project, and thanks for your help this far!

**Comment by sanjayananda** [comment]
Any news on this?

Used to use this app on mobile daily. It's been a few months without it, and no response since Mike stepped back. :(

Can I do anything to help this progress?

**Comment by mikehardy** [comment]
Hi there - no news as of yet, sorry. Just thinking about it quickly now while I'm looking at it, I'm not even sure how I would solve this...we reuse the WebView between question/answer (card flip) but we completely re-display the document inside of it.

The only thing I can think of is something like what @pigoz did with https://github.com/pigoz/anki-canvas - use local storage to share state across the question/answer flip. In this case you would want to store current scroll position I believe, which means you'd want some javascript that listened to scroll events and stored the scroll position when scrolling was stopped, then on answer load you'd want to see if there was a scroll position - use it then clear it if so, and you're set.

I am not saying this is easy, but I am thinking this is actually possible as a javascript solution and could be developed independent of the main project - possibly with the aid of the HTML / JS inspection powers available in the new alpha https://github.com/ankidroid/Anki-Android/wiki/Development-Guide#html-javascript-inspection

**Comment by sanjayananda** [comment]
Checking in - any update? Just downloaded AnkiDroid again to check on the issue, it's still present as described :(

**Comment by mikehardy** [comment]
This issue will be updated when/if anyone has time to get to it - sorry it's not faster but it's a volunteer project and we were wrapping up the 2.9 release

**Comment by sanjayananda** [comment]
Hey all, any news on this? It's been a few years 😅😔, so wanted to check in. Really missing getting to use Anki offline on a mobile device.

**Comment by mikehardy** [comment]
Ironically there was another comment on another issue recently related to the same chunk of code - I think we may remove it but I'm not sure, it needs testing: https://github.com/ankidroid/Anki-Android/issues/7825#issuecomment-738911823

**Comment by sanjayananda** [comment]
Thanks for the update! Glad it's still being considered 🙏

**Comment by BrayanDSO** [comment]
Issue too old with no activity nor someone else with the same issue + if it is not already fixed, it will with the new reviewer.

**Comment by ZornHadNoChoice** [comment]
There was a duplicate issue 2020, a [feature request on the forums in 2025](https://forums.ankiweb.net/t/keeping-the-viewport-in-the-same-place-when-showing-answers-in-cloze-completions/67725), and a new [feature request](https://forums.ankiweb.net/t/keep-screen-position-between-front-back-cards/70616) today.

The issue persists in AnkiDroid v2.25.0alpha2 both with the old and new study screens.

Interestingly, the preview screens don't have this issue.

**Comment by david-allison** [comment]
@ZornHadNoChoice I'll reopen, could I have a video of this occurring on the new study screen? That's unexpected.

**Comment by ZornHadNoChoice** [comment]
https://github.com/user-attachments/assets/4a5d4b4f-529b-473e-b0fc-59ee63778123

**Comment by BrayanDSO** [comment]
Isn't this the default Anki behavior of jumping to the `<hr id=answer>`?

**Comment by ZornHadNoChoice** [comment]
No. As you can see at the end of the video above, the template doesn't have an element with `id=answer`.

Anki Desktop keeps the scroll position when flipping the card if there's no element with `id=answer`. In particular, clozes cards don't scroll to the first active cloze, but they still preserve the scroll position.

**Comment by david-allison** [comment]
This is fixed in the new study screen.
